libs/Matchmaking/src/Peer/MemoryPeerDiscoveryTransport.cs

```diff
@@ -25,46 +26,54 @@ internal MemoryPeerDiscoveryMessage(MemoryPeerDiscoveryTransport sender, Guid st
 
     public class MemoryPeerDiscoveryTransport : IPeerDiscoveryTransport
     {
-        int ident_;
-        ConcurrentQueue<MemoryPeerDiscoveryMessage> incoming_ = new ConcurrentQueue<MemoryPeerDiscoveryMessage>();
+        private readonly int port_;
+        private readonly ConcurrentQueue<MemoryPeerDiscoveryMessage> incoming_ = new ConcurrentQueue<MemoryPeerDiscoveryMessage>();
 
-        //TODO extract into a factory so we can have independent transports
-        static volatile List<MemoryPeerDiscoveryTransport> instances_ = new List<MemoryPeerDiscoveryTransport>();
+        private static readonly ConcurrentDictionary<int, List<MemoryPeerDiscoveryTransport>> instances_ =
```

Comment:
Should we extract this static into a MemoryPeerDiscoveryNetwork?

Reply:
Probably not worth it - this is a test class, better to do the minimum work to make it function properly.